cpp/include/raft/core/memory_type.hpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
```

Comment:
I really like consolidating the different memory variations into it's own enum. I'm wondering if we should integrate / consolidate this further into the `raft::host_device_accessor` [here](https://github.com/rapidsai/raft/blob/2737d8d356918ddce351c040f52730a0dc5ab64a/cpp/include/raft/core/host_device_accessor.hpp#L32)? or maybe there's no value in doing that quite yet?

Reply:
I think it's certainly worth doing as a follow-up.